### PR TITLE
Added some tests and documentation

### PR DESCRIPTION
**Pull request by tdikland**
Added additional testing + documentation to some of the newer code.

### CODE REVIEW

File: delta-sharing/server/src/auth.rs [hunk 1]
```diff
@@ -1,37 +1,58 @@
...
+        AuthorizationLayer::new(authenticator)
+    }
 }
 
-impl<S, T: Clone + Send + Sync> Service<Request> for AuthenticationMiddleware<S, T>
+impl<S, T, R> Service<Request> for AuthenticationMiddleware<S, T>
 where
     S: Service<Request, Response = Response> + Send + 'static,
     S::Future: Send + 'static,
+    T: Authenticator<Recipient = R, Request = Request>,
```
Comment (tdikland) [inline delta-sharing/server/src/auth.rs:54]: Personally I like to model structs as "just data" and put the bounds on the impls.

File: delta-sharing/server/src/auth.rs [hunk 2]
```diff
@@ -1,37 +1,58 @@
...
+/// Authenticator that always marks the recipient as anonymous.
+#[derive(Clone)]
 pub struct AnonymousAuthenticator;
 
 impl Authenticator for AnonymousAuthenticator {
     type Request = Request;
     type Recipient = DeltaRecipient;
 
-    fn authenticate(&self, _: &Self::Request) -> Result<Self::Recipient, CoreError> {
+    fn authenticate(&self, _request: &Self::Request) -> Result<Self::Recipient, CoreError> {
```
Comment (roeap) [inline delta-sharing/server/src/auth.rs:21]: there is a slight nuance here and in practice there likely is no difference, but `_request` will create a variable binding that is dropped at the end of the scope, while for `_` this binding will never be created ...

Comment (roeap) [reply delta-sharing/server/src/auth.rs:21]: in the latter case it just tells the compiler not to complain about an unused variable.

Comment (tdikland) [reply delta-sharing/server/src/auth.rs:21]: Today I learned! 😃

Comment (tdikland) [reply delta-sharing/server/src/auth.rs]: Should be fixed now 🎉

**Review by roeap — APPROVED** [review]

**Pull request merged**